### PR TITLE
Add mount path into the default generated openapi.json spec (UI)

### PR DESCRIPTION
**Pull request by averche**
The current behaviour is to only add mount paths into the generated `opeanpi.json` spec if a `generic_mount_paths` flag is added to the request. This means that we would have to maintain two different `openapi.json` files, which is not ideal. The new solution in this PR is to add `{<mount>_mount_path}` into every path with a default value specified:

```diff
--    "/auth/token/accessors/": {
++    "/auth/{token_mount_path}/accessors/": {
      "parameters": [
++        {
++          "name": "token_mount_path",
++          "description": "....",
++          "in": "path",
++          "schema": {
++            "type": "string",
++          "default": "token"
++          }
++        }
...
     ],
```

Additionally, fixed the logic to generate the `operationId` (used to generate method names in the code generated from OpenAPI spec). It had a bug where the ID had `mountPath` in it.

### CODE REVIEW

File: ui/app/services/path-help.js [hunk 1]
```diff
@@ -178,32 +177,37 @@ export default Service.extend({
...
-      const path = Object.keys(help.openapi.paths)[0]; // do this or look at name
+      // help.openapi.paths is an array. Get the first unless the pathName is passed from caller
+      const path = pathName || Object.keys(help.openapi.paths)[0];
       const pathInfo = help.openapi.paths[path];
       const params = pathInfo.parameters;
       const paramProp = {};
 
       // include url params
       if (params) {
-        const { name, schema, description } = params[0];
```
Comment (hashishaw) [inline ui/app/services/path-help.js:195]: This is the crux of the change that fixes the UI issues. Before, we only cared about the first param for the path. Now that the backend mount path is also a path param, we instead iterate over all the path params, throw out `_mount_path` (because we already capture it as `backend`) and add the rest to the model

File: ui/app/services/path-help.js [hunk 2]
```diff
@@ -295,7 +299,8 @@ export default Service.extend({
   },
 
   registerNewModelWithProps(helpUrl, backend, newModel, modelName) {
-    return this.getProps(helpUrl, backend).then((props) => {
+    const pathName = modelName === 'model:generated-user-userpass' ? '/users/{username}' : '';
```
Comment (hashishaw) [inline ui/app/services/path-help.js:302]: This might not actually be needed, checking now

File: sdk/framework/openapi.go [hunk 3]
```diff
@@ -209,16 +209,16 @@ var (
 	altRootsRe       = regexp.MustCompile(`^\(([\w\-_]+(?:\|[\w\-_]+)+)\)(/.*)$`) // Pattern starting with alts, e.g. "(root1|root2)/(?P<name>regex)"
 	cleanCharsRe     = regexp.MustCompile("[()^$?]")                              // Set of regex characters that will be stripped during cleaning
 	cleanSuffixRe    = regexp.MustCompile(`/\?\$?$`)                              // Path suffix patterns that will be stripped during cleaning
-	nonWordRe        = regexp.MustCompile(`[^\w]+`)                               // Match a sequence of non-word characters
+	nonWordRe        = regexp.MustCompile(`[^a-zA-Z0-9]+`)                        // Match a sequence of non-word characters
```
Comment (dhuckins) [inline sdk/framework/openapi.go:212]: why was this changed? `\w` => `[a-zA-Z0-9_].` so just ignoring the `_` char?

**Review by dhuckins — APPROVED** [review]

File: sdk/framework/openapi.go (hunk 3 above)
Comment (averche) [reply sdk/framework/openapi.go:212]: Yes, this regex is negating the expression, so it's excluding the `_` char from matches. The reason we want this is because the regex is used to generate openapi `operationId`s  from paths. If a path has `_` characters in it, the old implementation would result in underscore characters sneaking into the `operationId`s while the new implementation changes them to CamelCaseExpressions. IIRC this was needed to addess the bug with mount paths being translated into `operationId`s.
